text/build_process.md

```diff
@@ -0,0 +1,146 @@
+# Build process and dependency management
+
+## Initial problems
+
+This RFC is intended to answer to these questions:
+> * Where should an adapter live?
+> * How can we prevent a change in Taxonomy to require an update of many repos?
+> * How can we make upgrades to the latest rust nightly easier (aka "rust up")?
+
+
+Currently, Foxbox is built with this dependency tree:
+```
+foxbox ─┬─> adapter_0   ─┬─> taxonomy
+        ├─> adapter_1   ─┤
+        ├─> adapter_n   ─┤
+        ├─> thinkerbell ─┤
+        └────────────────┘
+```
+Each of these dependencies lives in a separate git repository, and as a corollary, in a separate crate.
+
+We link these dependencies with each other thanks to git revisions. When cargo deals with git revisions, it will *always* build a crate with the specified revision. In other words, when `adapter_0` depends on taxonomy revision A but `foxbox` depends on revision B (which is more recent), cargo will first build revision A, then revision B. In addition to slowing down the compile time, we recurrently face compile errors due to revision. Moreover, if you're not familiar with either cargo dependency resolution or our dependency tree: you have to watch closely the cargo build steps to figure out revision A is being compiled again, and producing this error which should have been fixed by revision B.
```

Comment:
Note: we could use the output of `cargo metadata` to check this in travis. I'll file an issue for this. (maybe somebody did this somewhere already)